arabic-learning-app/src/app/page.tsx

```
@@ -181,8 +181,99 @@ export default function Home() {
     },
   ]);
 
+  // Function to sanitize JSON string by removing control characters and ensuring valid JSON
+ const sanitizeJsonString = (jsonStr: string): string => {
+   try {
+     // First attempt: Basic sanitization
+     let sanitized = jsonStr
+       .replace(/[\u0000-\u001F\u007F-\u009F]/g, "") // Remove control characters
+       .replace(/\n/g, "\\n") // Properly escape newlines
+       .replace(/\r/g, "\\r") // Properly escape carriage returns
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)